### PR TITLE
 Optimizing code & add some notes

### PR DESCRIPTION
**Pull request by af913337456**
## Description

1. Some obvious duplicate calculations have been removed.
2. In order to better understand the tipping-related code, some comment are added to the output code of the tips.

## Type of change

- [x] Optimizing code

## Brief of your solution

Repeated computational code, which assigns values to variables only once, is then used publicly.

### CODE REVIEW

**Comment by qizheng09** [comment]
Thanks for your PR. But, its better to rebase your master to upstream : ）.

**Comment by af913337456** [comment]
@qizheng09 Rebased , is that so?

File: utxo/utxo.go [hunk 1]
```diff
@@ -987,7 +989,7 @@ func (uv *UtxoVM) doTxInternal(tx *pb.Transaction, batch kvdb.Batch) error {
 	}
 	for offset, txOutput := range tx.TxOutputs {
 		addr := txOutput.ToAddr
-		if bytes.Equal(addr, []byte(FeePlaceholder)) {
+		if bytes.Equal(addr, []byte(FeePlaceholder)) { // 小费类型的 output 不在这里处理
```
Comment (yucaowang) [inline utxo/utxo.go]: Why do you add such comments here?

File: utxo/utxo.go [hunk 2]
```diff
@@ -921,7 +923,7 @@ func (uv *UtxoVM) subBalance(addr []byte, delta *big.Int) {
 func (uv *UtxoVM) payFee(tx *pb.Transaction, batch kvdb.Batch, block *pb.InternalBlock) error {
 	for offset, txOutput := range tx.TxOutputs {
 		addr := txOutput.ToAddr
-		if !bytes.Equal(addr, []byte(FeePlaceholder)) {
+		if !bytes.Equal(addr, []byte(FeePlaceholder)) { // 排除掉不是小费的 output
```
Comment (yucaowang) [inline utxo/utxo.go]: ditto

**Comment by yucaowang** [comment]
> @qizheng09 Rebased , is that so?

means update your branch to avoid merged changes

File: utxo/utxo.go (hunk 2 above)
Comment (af913337456) [reply utxo/utxo.go]: According to my understanding of the code, "output" with the address pointing to "FeePlaceholder" is a consumption for miners. Moreover, the "payFee" function tips the miners, and the line of this code "if! Bytes. Equal(addr, [] byte (FeePlaceholder)" obviously filters out the "output" of the receiving address that is not"FeePlaceholder".So I added a comment: “排除掉不是小费的 output”,Is there any mistake in my understanding? Thank you.

**Comment by af913337456** [comment]
@yucaowang After added the git url of  master branch,i have ran this cmd "git merge upstream/master" and got those msg "All files are up-to-date".Has it been merged the branch to avoid merged changes？thanks.

**Comment by af913337456** [comment]
@yucaowang @qizheng09 btw, as shown in the image below, is my way of merging the main branches wrong?thanks.

image

**Comment by qizheng09** [comment]
@af913337456 Firstly, please `git fetch upstream` .

File: utxo/utxo.go (hunk 2 above)
Comment (yucaowang) [reply utxo/utxo.go]: Please don't add personal comments on existing code, the code is clear enough. :)

**Comment by af913337456** [comment]
@qizheng09 thanks,it work

File: utxo/utxo.go (hunk 2 above)
Comment (af913337456) [reply utxo/utxo.go]: yep, i remove it

File: utxo/utxo.go [hunk 3]
```diff
@@ -816,12 +817,14 @@ func (uv *UtxoVM) sortUnconfirmedTx() (map[string]*pb.Transaction, TxGraph, map[
...
 		}
 	}
-	if len(txMap) > 0 {
-		avgDelay := totalDelay / int64(len(txMap)) //平均unconfirm滞留时间
-		uv.xlog.Info("average unconfirm delay", "micro-senconds", avgDelay/1e6, "count", len(txMap))
-		uv.avgDelay = avgDelay / 1e6
+	txMapSize := int64(len(txMap))
+	if txMapSize > 0 {
+		avgDelay := totalDelay / txMapSize //平均unconfirm滞留时间
+		microSec := avgDelay / 1e6
```
Comment (ToWorld) [inline utxo/utxo.go]: If an expression is not used multiple times, no need to correct it.

Comment (yucaowang) [reply utxo/utxo.go]: @af913337456 , any response to this comment?

Comment (af913337456) [reply utxo/utxo.go]: @yucaowang What's your suggestion about this? I'd like to know about it.

Comment (af913337456) [reply utxo/utxo.go]: @yucaowang In my opinion, my modifications include making the code look better.

**Pull request merged**